example/run_py_example.sh

```diff
@@ -16,8 +16,16 @@ base_dir=$(cd "`dirname "$loc"`" && pwd)
 
 python_cmd=$1
 
+export PYTHONPATH=${PYTHONPATH:-""}
 export PYTHONPATH="$base_dir/../support-lib/py:$base_dir/../build_py/cffi:$base_dir/generated-src/python:$PYTHONPATH"
-export DYLD_LIBRARY_PATH=$base_dir/../build_py/cffi:$DYLD_LIBRARY_PATH
-(cd "$base_dir/../build_py/cffi" \
+# The use of install_name_tool below is a hack to force MacOS to know where to load libtextsort_py.dylib,
+# needed because DYLD_LIBRARY_PATH is now ignored by System Integrity Protection.  A real installed which
+# used a dylib like this would need a cleaner solution based on @rpath or somesuch thing.  The file existence
+# checks before the commands look for the two different .so names produced for CPython 2 or 3.
+#export DYLD_LIBRARY_PATH=$base_dir/../build_py/cffi:$DYLD_LIBRARY_PATH
+cffi_dir=$(cd "$base_dir/../build_py/cffi" && pwd)
+(cd "$cffi_dir" \
         && "$python_cmd" ../../example/generated-src/cffi/pycffi_lib_build.py ../../support-lib/cwrapper/wrapper_marshal.h $(ls ../../example/generated-src/cwrapper/*.h) \
+        && echo && ([ ! -f PyCFFIlib_cffi.so ] || install_name_tool -change /usr/local/lib/libtextsort_py.dylib "$cffi_dir/libtextsort_py.dylib" PyCFFIlib_cffi.so) \
```

Comment:
I'm not quite familiar with `install_name_tool`.
Looks like if the `install_name_tool` has been executed, the test for `-f PyCFFIlib_cffi.so` would then fail?

I'm wondering if there is a better alternative for this testing, the `-f` test seems not quite robust.